lxd-user/lxd.go

```diff
@@ -0,0 +1,255 @@
+package main
+
+import (
+	"encoding/base64"
+	"fmt"
+	"os"
+	"os/user"
+	"path/filepath"
+	"strings"
+
+	"github.com/lxc/lxd/client"
+	"github.com/lxc/lxd/lxd/revert"
+	"github.com/lxc/lxd/lxd/util"
+	"github.com/lxc/lxd/shared"
+	"github.com/lxc/lxd/shared/api"
+)
+
+func lxdIsConfigured(client lxd.InstanceServer) (bool, error) {
+	// Look for networks.
+	networks, err := client.GetNetworkNames()
+	if err != nil {
+		return false, fmt.Errorf("Failed to list networks: %w", err)
+	}
+	if !shared.StringInSlice("lxdbr0", networks) {
+		// Couldn't find lxdbr0.
+		return false, nil
+	}
+
+	// Look for storage pools.
+	pools, err := client.GetStoragePoolNames()
+	if err != nil {
+		return false, fmt.Errorf("Failed to list storage pools: %w", err)
+	}
+
+	if !shared.StringInSlice("default", pools) {
+		// No storage pool found.
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func lxdInitialConfiguration(client lxd.InstanceServer) error {
```

Comment:
Should this function get a reverter?

Reply:
I thought of that but I don't think it's needed. The reason is that we will call `lxdInitialConfiguration` again until we have both a storage pool and a network setup. So it failing partway shouldn't actually be a problem as re-running it will yield the right result.

There's also the fact that I'm never too keen on having code that automatically calls DeleteStoragePool and DeleteNetwork :)